apps/prairielearn/src/sync/fromDisk/assessments.js

```diff
@@ -315,6 +325,71 @@ module.exports.sync = async function (courseId, courseInstanceId, assessments, q
       }
     });
   }
+
+  /** @type {Set<string>} */
+  const importedQids = new Set();
+  /** @type {Map<string, string[]>} */
+  const importedQidAssessmentMap = new Map();
+  Object.entries(assessments).forEach(([tid, assessment]) => {
+    if (!assessment.data) return;
+    (assessment.data.zones || []).forEach((zone) => {
+      (zone.questions || []).forEach((question) => {
+        let qids = question.alternatives
+          ? question.alternatives.map((alternative) => alternative.id)
+          : [];
+        if (question.id) {
+          qids.push(question.id);
+        }
+        qids.forEach((qid) => {
+          if (qid[0] === '@') {
+            importedQids.add(qid);
+            let tids = importedQidAssessmentMap.get(qid);
+            if (!tids) {
+              tids = [];
+              importedQidAssessmentMap.set(qid, tids);
+            }
+            tids.push(tid);
+          }
+        });
+      });
+    });
+  });
+
+  if (importedQids.size > 0) {
+    let result = await sqldb.queryOneRowAsync(sql.get_course_info, { course_id: courseId });
+    if (!result.rows[0].question_sharing_enabled && config.checkSharingOnSync) {
+      for (let qid of importedQids) {
+        importedQidAssessmentMap.get(qid)?.forEach((tid) => {
+          infofile.addError(
+            assessments[tid],
+            `You have attempted to import a question with '@', but question sharing is not enabled for your course.`
+          );
+        });
+      }
+    }
+  }
+
+  const importedQuestions = await sqldb.queryAsync(sql.get_imported_questions, {
+    course_id: courseId,
+    imported_qids: Array.from(importedQids),
+  });
+  for (let row of importedQuestions.rows) {
+    questionIds['@' + row.sharing_name + '/' + row.qid] = row.id;
+  }
+  let missingQids = Array.from(importedQids).filter((qid) => !(qid in importedQuestions));
```

Comment:
AFAICT `importedQuesitons` is an object that looks like `{ rows: [ ... ] }`; how does this work?

Reply:
nice catch! This in fact, does not work.

I know it did at some point, something must have gotten lost in a refactor. Anyway, I will fix it and add a test to make sure it stays fixed.

Reply:
done